### PR TITLE
Correctly fail EventLoopFuture for unsupported operations on ServerSo…

### PR DESCRIPTION
**Pull request by normanmaurer**
…cketChannel

Motivation:

ServerSocketChannel should just fail the EventLoopFuture for unsupported operations and not crash.

Modifications:

- Correctly fail the EventLoopFuture for unsupported operations
- Add unit test.

Result:

No more crashes when users invoke unsupported operations.

### CODE REVIEW

**Comment by normanmaurer** [comment]
Fixes https://github.com/apple/swift-nio/issues/222

File: Sources/NIO/SocketChannel.swift [hunk 1]
```diff
@@ -436,6 +436,19 @@ final class ServerSocketChannel: BaseSocketChannel<ServerSocket> {
...
+        promise?.fail(error: ChannelError.operationUnsupported)
+    }
+
+    override func markFlushPoint(promise: EventLoopPromise<Void>?) {
+        promise?.fail(error: ChannelError.operationUnsupported)
+    }
+
+    override func writeToSocket() throws -> OverallWriteResult {
+        // We just return .writtenCompletely as there is never anything to write and this is the cheapest thing to do here.
+        return .writtenCompletely
```
Comment (Lukasa) [inline Sources/NIO/SocketChannel.swift]: Does this function ever get called? I think we can `preconditionFailure` here on the grounds that we should never enter this.

Comment (normanmaurer) [reply Sources/NIO/SocketChannel.swift]: @Lukasa it actually was from `flush0()`. That said I just override `flush0()` now for `ServerSocketChannel`.

**Comment by normanmaurer** [comment]
@Lukasa PTAL again

**Pull request merged**